src/lib/node/net.js

```diff
@@ -656,16 +656,36 @@ foam.CLASS({
       class: 'Boolean',
       name: 'followRedirect',
       value: true
+    },
+    {
+      name: 'urlLib',
+      value: require('url')
     }
   ],
 
   methods: [
     function fromUrl(url) {
-      var data = require('url').parse(url);
-      this.protocol = data.protocol.slice(0, -1);
-      this.hostname = data.hostname;
+      var data = this.urlLib.parse(url);
+      if ( data.protocol ) this.protocol = data.protocol.slice(0, -1);
+      if ( data.hostname ) this.hostname = data.hostname;
       if ( data.port ) this.port = data.port;
-      this.path = data.path;
+      if ( data.path ) this.path = data.path;
+
+      return this;
+    },
+
+    function copyUrlFrom(other) {
```

Comment:
Well, for one thing, we don't want to copy everything, only the URL-related stuff. So using copyFrom would be something like

```js
foo.copyFrom({
  protocol: bar.protocol,
  /* ... */
})
```

For another thing, port is copied conditionally, which further mucks with a `copyFrom()` call. That's why I wanted to consolidate the careful-imperative-copy in to a helper method.